pages/docs/tutorials/kafka/managing-schemas-using-schema-registry.md

```diff
@@ -0,0 +1,103 @@
+
+While several Schema Registry implementations exist, you will use the [Apicurio Registry](https://www.apicur.io/registry/) in this tutorial. Apicurio Registry is a popular open-source Schema Registry implementation that supports multiple serialization formats and facilitates schema management for diverse data in distributed systems. You will be using the Apicurio Registry in combination with Avro, which is a language-neutral data serialization system.
+
+## Prerequisites
+Install [Docker](https://docs.docker.com/engine/install/) from the official website.
+
+## Steps
+1. Start Apicurio Registry locally using below docker command:
+   
+```
+{`docker run --env CORS_ALLOWED_ORIGINS=* -it -p 8080:8080 apicurio/apicurio-registry-mem:2.5.8.Final`}
+```
+
+2. Once your local instance of Apicurio Registry is running, you can upload your Avro schema to it. Open new terminal window and create an Avro schema artifact using the following command:
+   
+```
+curl \
+http://localhost:8080/apis/registry/v2/groups/my-group/artifacts \
+-X POST  \
+-H "Content-Type: application/json; artifactType=AVRO" \
+-H "X-Registry-ArtifactId: UserSignedUp" \
+--data @- << EOF
+{
+  "type": "record",
+  "name": "UserSignedUp",
+  "namespace": "com.company",
+  "doc": "User sign-up information",
+  "fields": [
+    {
+      "name": "userId",
+      "type": "int"
+    },
+    {
+      "name": "userEmail",
+      "type": "string"
+    }
+  ]
+}
+EOF
+```
+
+<Remember>
+Download your Avro schema by visiting the following URL: 
+http://localhost:8080/apis/registry/v2/groups/my-group/artifacts/UserSignedUp.
+</Remember>
+
+3. An alternative to having the schema in a separate file as you learned in the [previous tutorial](/docs/tutorials/kafka/configure-kafka-avro) where you used the existing Avro schema in your AsyncAPI document, would be to store it in the registry. After uploading your Avro schema, remove the schema from your document, and replace it with the following:
+$ref: {url}
+Replace {url} with the url from the previous step.
+
+```
+asyncapi: 3.0.0
+info:
+  title: User Signup API
+  version: 1.0.0
+  description: The API notifies you whenever a new user signs up in the application.
+servers:
+  kafkaServer:
+    host: test.mykafkacluster.org:8092
+    description: Kafka Server
+    protocol: kafka
+operations:
+  onUserSignedUp:
+    action: receive
+    channel:
+      $ref: '#/channels/userSignedUp'
+channels:
+  userSignedUp:
+    description: This channel contains a message per each user who signs up in our application.
+    address: user_signedup
+    messages:
+      userSignedUp:
+        $ref: '#/components/messages/userSignedUp'
+components:
+  messages:
+    userSignedUp:
+      payload:
+        schemaFormat: 'application/vnd.apache.avro+json;version=1.9.0'
+        schema:
+          $ref: http://localhost:8080/apis/registry/v2/groups/my-group/artifacts/UserSignedUp
+```
```

Comment:
I don't think your code snippet is a valid document because [Studio is marking errors](https://studio.asyncapi.com/?base64=YXN5bmNhcGk6IDMuMC4wCmluZm86CiAgdGl0bGU6IFVzZXIgU2lnbnVwIEFQSQogIHZlcnNpb246IDEuMC4wCiAgZGVzY3JpcHRpb246IFRoZSBBUEkgbm90aWZpZXMgeW91IHdoZW5ldmVyIGEgbmV3IHVzZXIgc2lnbnMgdXAgaW4gdGhlIGFwcGxpY2F0aW9uLgpzZXJ2ZXJzOgogIGthZmthU2VydmVyOgogICAgaG9zdDogdGVzdC5teWthZmthY2x1c3Rlci5vcmc6ODA5MgogICAgZGVzY3JpcHRpb246IEthZmthIFNlcnZlcgogICAgcHJvdG9jb2w6IGthZmthCm9wZXJhdGlvbnM6CiAgb25Vc2VyU2lnbmVkVXA6CiAgICBhY3Rpb246IHJlY2VpdmUKICAgIGNoYW5uZWw6CiAgICAgICRyZWY6ICcjL2NoYW5uZWxzL3VzZXJTaWduZWRVcCcKY2hhbm5lbHM6CiAgdXNlclNpZ25lZFVwOgogICAgZGVzY3JpcHRpb246IFRoaXMgY2hhbm5lbCBjb250YWlucyBhIG1lc3NhZ2UgcGVyIGVhY2ggdXNlciB3aG8gc2lnbnMgdXAgaW4gb3VyIGFwcGxpY2F0aW9uLgogICAgYWRkcmVzczogdXNlcl9zaWduZWR1cAogICAgbWVzc2FnZXM6CiAgICAgIHVzZXJTaWduZWRVcDoKICAgICAgICAkcmVmOiAnIy9jb21wb25lbnRzL21lc3NhZ2VzL3VzZXJTaWduZWRVcCcKY29tcG9uZW50czoKICBtZXNzYWdlczoKICAgIHVzZXJTaWduZWRVcDoKICAgICAgcGF5bG9hZDoKICAgICAgICBzY2hlbWFGb3JtYXQ6ICdhcHBsaWNhdGlvbi92bmQuYXBhY2hlLmF2cm8ranNvbjt2ZXJzaW9uPTEuOS4wJwogICAgICAgIHNjaGVtYToKICAgICAgICAgICRyZWY6IGh0dHA6Ly9sb2NhbGhvc3Q6ODA4MC9hcGlzL3JlZ2lzdHJ5L3YyL2dyb3Vwcy9teS1ncm91cC9hcnRpZmFjdHMvVXNlclNpZ25lZFVw).

<img width="1631" alt="Screenshot 2024-01-29 at 7 00 00 PM" src="https://github.com/asyncapi/website/assets/19964402/1ead8f71-6df8-4bd0-af95-27ca42a0b4c1">

Reply:
The actual code snippet is the one from Joy's tutorial which works perfectly. 
![image](https://github.com/asyncapi/website/assets/84087089/bb58bb13-9974-4272-b0fb-82c0539d70ce)

Here, we've removed the schema since it has already been uploaded by whoever is following this tutorial, and replaced it with $ref that points to the uploaded schema. I think that Studio is marking errors because you don't have the schema uploaded.